server/server.go

```diff
@@ -537,112 +654,214 @@ var (
-		switch c.Request.Method {
-		case http.MethodGet, http.MethodDelete:
-			var req REQUEST
-			response, apiError = handlerFn(c.Request.Context(), req)
-			break
-		case http.MethodPost, http.MethodPut, http.MethodPatch:
-			req, err := extractRequest[REQUEST](c, requestValidator)
-			if err != nil {
-				apiError = err
-				break
-			}
-			response, apiError = handlerFn(c.Request.Context(), req)
-			break
-		default:
-			apiError = serr.NewErrorResponseFromApiError(errMethodNotAllowed)
-			break
+		var req *REQUEST
+		if r, ok := onExtractRequestBodyAndParameters(c, extractRequestArgsFn, logger, func(r *REQUEST) bool { return onValidateRequest(c, r, logger, requestValidator, extensions) }); !ok {
+			return
+		} else {
+			req = r
 		}
 
+		response, apiError := handlerFn(c.Request.Context(), *req)
 		if apiError != nil {
 			writeAndLogApiErrorThenAbort(c, apiError, logger)
 			return
 		}
 
-		var r RESPONSE
-		responseType := reflect.TypeOf(r)
-		if response == nil || reflect.ValueOf(&response.Body).Elem().IsZero() {
-			if responseType != nil && responseType == voidType {
-				c.Status(http.StatusNoContent)
-				_, _ = c.Writer.Write([]byte{})
-				return
-			} else {
-				writeAndLogApiErrorThenAbort(c, serr.NewErrorResponseFromApiError(
-					errServerFailedToProduceExpectedResponse,
-					serr.WithErrorMessage("The handler returned a nil response or nil response.Body but the response type was not server.Void, your handler should return *server.Response[server.Void] if you want to have no response body, else you must return a non nil response object."),
-					serr.WithStackTraceLoggingBehavior(serr.ForceNoStackTrace),
-				), logger)
-				return
-			}
-		}
+		onHandleResponse(c, response, logger, handler, apiError)
+	}
+}
 
-		statusCode := http.StatusOK
-		if handler.StatusCode != 0 {
-			statusCode = handler.StatusCode
+func onHandleResponse[RESPONSE any](c *gin.Context, response *Response[RESPONSE], logger *zap.SugaredLogger, handler *handlerDTO, apiError serr.Error) {
+	var r RESPONSE
+	responseType := reflect.TypeOf(r)
+	if response == nil || reflect.ValueOf(&response.Body).Elem().IsZero() {
+		if responseType != nil && responseType == voidType {
+			c.Status(http.StatusNoContent)
+			_, _ = c.Writer.Write([]byte{})
+			return
+		} else {
+			writeAndLogApiErrorThenAbort(c, serr.NewErrorResponseFromApiError(
+				errServerFailedToProduceExpectedResponse,
+				serr.WithErrorMessage("The handler returned a nil response or nil response.Body but the response type was not server.Void, your handler should return *server.Response[server.Void] if you want to have no response body, else you must return a non nil response object."),
+				serr.WithStackTraceLoggingBehavior(serr.ForceNoStackTrace),
+			), logger)
+			return
 		}
-		if response.StatusCode != 0 {
-			statusCode = response.StatusCode
+	}
+
+	statusCode := http.StatusOK
+	if handler.StatusCode != 0 {
+		statusCode = handler.StatusCode
+	}
+	if response.StatusCode != 0 {
```

Comment:
Are we hiding any issues by overwriting the handler statusCode here?

Reply:
that is only sucess flow - should there be an error - this code will not be executed. by default - 500 status code is returned, unless returned error specifies otherwise.